homeassistant/components/plugwise/sensor.py

```diff
@@ -351,29 +380,15 @@ def _async_process_data(self) -> None:
             self.async_write_ha_state()
             return
 
-        if data.get(self._sensor) is not None:
-            self._attr_native_value = data[self._sensor]
-            self._attr_icon = CUSTOM_ICONS.get(self._sensor, self.icon)
-
+        self._attr_native_value = data.get(self.entity_description.key)
```

Comment:
It looks like it didn't update before if the value was `None`.  Is there any risk that sometimes the data doesn't come though in an update so it needs to wait for the next one?

Reply:
I haven't observed such a case. I think the existing logic was incorrect for that matter as well, because it means the value sticks to the previous value (instead of becoming unknown).

In the next PR, the data will be provided via the data update coordinator (including a library bump). At that point, the data update coordinator would actually protect against an invalid/unexpected response (making the devices unavailable).

Reply:
It looked like incorrect logic to me as well ... but with the number of flakey apis I've seen you never know 🙈